draft-ietf-dnssd-privreq.xml

```diff
@@ -817,7 +824,128 @@ This draft does not require any IANA action.
+    <author initials='H.' surname='Krawczyk' fullname='Hugo Krawczyk'>
+      <organization>EE Department, Technion, Haifa, Israel, and IBM T.J. Watson Research Center</organization>
+    </author>
+    <date year="2003"/>
+  </front>
+</reference>
+
 </references>  
 
+  <section title="Survey of Implementations">
+
+    <t>This section surveys several private service discovery designs in the context of the threat model
+      detailed above.</t>
+
+    <section title="DNS-SD Privacy Extensions">
+      <t>Huitema and Kaiser <xref target="I-D.huitema-dnssd-privacy"></xref> decompose 
+      private service discovery into two stages: (1) identify specific peers offering 
+      private services, and (2) issue unicast DNS-SD queries to those hosts after 
+      connecting over TLS using a previously agreed upon pre-shared key (PSK), or 
+      pairing key. Any out-of-band pairing mechanism will suffice for PSK establishment, 
+      though the authors specifically mention <xref target="I-D.ietf-dnssd-pairing"></xref> 
+      as the pairing mechanism. Step (1) is done by broadcasting "private instance names"
+      to local peers, using service-specific pairing keys. A private instance name N' 
+      for some service with name N is composed of a unique nonce r and commitment to r using 
+      N_k. Commitments are constructed by hashing N_k with the nonce. Only owners of N_k 
+      may verify its correctness and, upon doing so, answer as needed. The draft recommends 
+      randomizing hostnames in SRV responses to minimize likability to specific hosts. 
+      Note that this does not prevent fingerprinting and tracking using that hostname.</t>
+
+      <t>After discovering its peers, a node will directly connect to each device using 
+        TLS, authenticated with a PSK derived from each associated pairing key, and 
+        issue DNS-SD queries per usual. DNS messages are formulated as per 
+        <xref target="RFC7858"></xref>.</t>
+
+      <t>As an optimization, the authors recommend that each nonce be deterministically 
+        derived based on time so that commitment proofs may be precomputed asynchronously. 
+        This avoids O(N*M) computation, where N is the number of nodes in a local network 
+        and M is the number of per-node pairings.</t>
+
+      <t>
+        This system has the following properties:
+        <list style='numbers'>
+          <t>Symmetric work load: clients and servers can pre-compute private instance names
+            as a function of their pairing secret and predictable nonce.</t>
+          <t>Mutual identity privacy: Both client and server identities are hidden from 
+            active and passive attackers that do not subvert the pairing process.</t>
+          <t>No client set size hiding: The number of private instance names
+          reveals the number of unique pairings a server has with its clients.</t>
```

Comment:
Actually, there is a way to mitigate that, by padding the list with randomly drawn instances. I think we do discuss it in the draft. It leads to the classic tradeoff between padding and fingerprinting, but it is certainly doable.

Reply:
I'm less enthused about that approach, though I'll add text clarifying that it's possible.